server/spinwick.go

```diff
@@ -583,8 +583,32 @@ func checkMMPing(ctx context.Context, client *mattermostModel.Client4) error {
 	}
 }
 
+// Given a string, returns an abbreviation of that string
+// All repositories under the mattermost organization follow a pattern of words-separated-by-dashes (I checked)
+// If there are no '-' characters, the first 4 characters of the repository name are returned to avoid collisions
+func getRepoNameAbbreviation(repoName string) string {
+	abbreviation := ""
+	segments := strings.Split(repoName, "-")
+	for _, segment := range segments {
+		abbreviation = abbreviation + string(segment[0])
+	}
+
+	//If there were no - separators in the repoName, use the first 4 characters
+	if len(abbreviation) == 1 {
+		return repoName[0:4]
```

Comment:
So, I know that this function should only be called on longer repo names, but this could technically panic. If you moved the length check to this function you could guard against that.

Reply:
Fixed in that this no longer exists anymore